doc/kubeletConfig-remediations.md

```diff
@@ -0,0 +1,104 @@
+# KubeletConfig Remediation proposed flow design
+
+This page describes the design and implementation of the KubeletConfig remediation templating support in the compliance operator.
+
+## Goals
+We want to accomplish:
+   * The remediation content can have `KubeletConfig` as payload
+   * We need to update the `parse_arf_result` to be able to take `KubeletConfig`
+   * The remediation controller will be able to merge KubeletConfig remediation into the correct KubeletConfig that is currently being used by the cluster
+
+## Add CaC `KubletConfig` remediation content 
+In CaC content, we are going to add `KubletConfig` remediation besides `MachineConfig` that we currently have.
+
+We will have remdiation that has object payload to be `KubletConfig`. As a exmaple below:
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: co-remediation
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      custom-kubelet: compliance-remediaiton
```

Comment:
will we need this selector? Isn't this something that CO will fill?

Reply:
So in order to have `KubeletConfig` to be processed and render a machine config, it need to have some sort of  matchLabel to match machineConfigPool, if a admin didn't use any custom`KubeletConfig`, we are going to add one here.

Reply:
Let's add these details to the doc

Reply:
Sounds good